### ISSUE TITLE
[REQ][PHP] Follow PSR-12 Coding Style Guide

### ISSUE THREAD

**Issue by ybelenko** [OpenAPITools/openapi-generator#4726]
### Is your feature request related to a problem? Please describe.

From [PSR-2 Coding Style Guide](https://www.php-fig.org/psr/psr-2/):
> **Deprecated** - As of 2019-08-10 PSR-2 has been marked as deprecated. [PSR-12](http://www.php-fig.org/psr/psr-12/) is now recommended as an alternative.

## Describe the solution you'd like

1. [x] ~~Change PHP style guide link at [Guidelines For Contributing](https://github.com/OpenAPITools/openapi-generator/blob/master/CONTRIBUTING.md#style-guide) to [PSR-12 Extended Style Guide](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-12-extended-coding-style-guide.md)~~
2. [ ] All PHP generators which follow `PSR-2` coding guide should follow new rules from `PSR-12`. Tool `PHP_CodeSniffer` already supports `PSR-12` ruleset from [`3.5.0`](https://github.com/squizlabs/PHP_CodeSniffer/releases/tag/3.5.0) version. If your generator contains `PHP_CodeSniffer` package then migrate to `^3.5` version and use ruleset like `phpcs --standard=PSR12 /path/to/code`(or adjust `phpcs.xml` in your template).
3. [ ] Adjust all other PHP linters and prettifiers to follow `PSR-12`.

## Additional context

Question to PHP members of technical committee: 
**As soon as required PHP version is 7.1 or higher should we add type annotations to PHP templates? I mean `int $bar = 0;`**

cc @jebentier (2017/07), @dkarlovi (2017/07), @mandrean (2017/08), @jfastnacht (2017/09), @ackintosh (2017/09), @renepardon (2018/12)

**Comment by ybelenko** [comment]
@wing328 Don't you think we should keep it open until all PHP generators adjust their linters to comply PSR-12? Or they can leave a message about current code formatting status at least?

**Comment by wing328** [comment]
Right, just reopened.

**Comment by ybelenko** [comment]
I think we can finally add PSR-12 ruleset to CS-Fixer configs.
[PSR-12 Support - Issue 4502 - PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4502)

It's not fully supported, but since this ruleset mentioned in the main doc I think we can migrate already.

**Comment by ybelenko** [comment]
@Articus should we add [friendsofphp/php-cs-fixer](https://packagist.org/packages/friendsofphp/php-cs-fixer) package to `php-dt(beta)` and`php-mezzio-ph`? What do you think.

**Comment by Articus** [comment]
> @Articus should we add [friendsofphp/php-cs-fixer](https://packagist.org/packages/friendsofphp/php-cs-fixer) package to `php-dt(beta)` and`php-mezzio-ph`? What do you think.

In my opinion - nope. This package is needed neither to run generated code nor to adjust generated code for existing project integration. So there is no need to bloat generated `composer.json` with it.

And by the way PSR-2 and PSR-12 are rather controversial "standards" - otherwise things like [PSR-2-R](https://github.com/php-fig-rectified/fig-rectified-standards) would not appear.

**Comment by ybelenko** [comment]
> And by the way PSR-2 and PSR-12 are rather controversial "standards" - otherwise things like [PSR-2-R](https://github.com/php-fig-rectified/fig-rectified-standards) would not appear.

Yes, I noticed it earlier. But we need to stick to some coding style in PHP anyway(originally it was PSR-2 in this repo, now it's PSR-12). It's easier to contribute when you can just follow some standard and it's easier to review changes. Beside we can turn off controversial rules in PHP CS Fixer anytime.

As soon as you're the major contributor and author of generators mentioned above then I don't insist. Will add CS Fixer to rest of PHP generators still.
